### PR TITLE
Added chapter Signing Packages 

### PR DESCRIPTION
**Pull request by Daskyleia**
Solving bz: https://bugzilla.redhat.com/show_bug.cgi?id=1564079

### CODE REVIEW

File: source/advanced-topics.adoc [hunk 1]
```diff
@@ -4,6 +4,187 @@
...
+* xref:Replacing-a-Package-Signature[Replacing the signature on an already existing package].
+
+* xref:Build-time-signing[Signing a package at build-time].
+
+[[Adding-a-Signature-to-a-Package]]
+=== Adding a Signature to a Package
+
+In most cases packages are build without a signature. The signature is added just before release of the package.
+
+In order to add another signature to the package use the --addsign option.
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: ``package use the --addsign option.`` > ``package, use the `--addsign` option.``

File: source/advanced-topics.adoc [hunk 2]
```diff
@@ -4,6 +4,187 @@
...
+Having more than one signature makes it possible to record the package's path of ownership
+from the package builder to the end-user.
+
+.Example:
+----
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Remove the literal block.

File: source/advanced-topics.adoc [hunk 3]
```diff
@@ -4,6 +4,187 @@
...
+As an example, a division of a company creates a package and signs it with the division's key. The company's headquarters then checks the package's signature and adds the corporate signature to the package, stating that the signed package is authentic.
+
+With two signatures, the package makes its way to a retailer. The retailer checks the signatures and, if they check out, adds their signature as well.
+
+The package now makes its way to a company that wishes to deploy the package. After checking every signature on the package, they know that it is an authentic copy, unchanged since it was first created. Depending on the deploying company's internal controls, they may choose to add their own signature, to reassure their employees that the package has received their corporate approval.
+----
+
+The output from the --addsign option:
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: ``from the `--addsign` option:``

File: source/advanced-topics.adoc [hunk 4]
```diff
@@ -4,6 +4,187 @@
...
+
+[source,bash]
+----
+# rpm --addsign blather-7.9-1.i386.rpm
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Should this be run as root (#) or as a normal user ($)? Same for other commands.

File: source/advanced-topics.adoc [hunk 5]
```diff
@@ -4,6 +4,187 @@
...
+
+To check the signatures of a package with multiple signatures:
+
+[source,bash]
+----
+# rpm --checksig blather-7.9-1.i386.rpm
+blather-7.9-1.i386.rpm: size pgp pgp md5 OK
+----
+
+The two "pgp" strings in the output of the "rpm --checksig" command show that the package has been signed twice.
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Use `literal` instead of "double quotes".

File: source/advanced-topics.adoc [hunk 6]
```diff
@@ -4,6 +4,187 @@
...
+
+RPM makes it possible to add the same signature multiple times. The --addsign option does not check for multiple identical signatures.
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: "The `--addsign` option"

File: source/advanced-topics.adoc [hunk 7]
```diff
@@ -4,6 +4,187 @@
...
+# rpm --checksig blather-7.9-1.i386.rpm
+blather-7.9-1.i386.rpm: size pgp pgp pgp pgp md5 OK
+----
+
+The output of the "rpm --checksig" command displays four signatures.
+
+[[Replacing-a-Package-Signature]]
+=== Replacing a Package Signature
+
+To change the public key without having to rebuild each package, use the "--resign" option.
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Use literal instead of "double quotes".

File: source/advanced-topics.adoc [hunk 8]
```diff
@@ -4,6 +4,187 @@
...
+…
+Generating signature: 1002
+Wrote: /usr/src/redhat/SRPMS/blather-7.9-1.src.rpm
+----
+
+The "Generating signature" message appears in both the binary and source packaging sections. The number following the message indicates that the signature added was created using PGP.
+
+[NOTE]
+--
+When using the "--sign" option for "rpmbuild", use only -bb or -ba options for package building.
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Use literal for all option names.

File: source/advanced-topics.adoc [hunk 9]
```diff
@@ -4,6 +4,187 @@
...
+--
+
+To verify the signature of a package, use the rpm command with "--checksig" option. For example:
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Use literal for all command names.

File: source/advanced-topics.adoc [hunk 10]
```diff
@@ -4,6 +4,187 @@
...
+
+[source,bash]
+----
+# rpm --checksig blather-7.9-1.i386.rpm
+blather-7.9-1.i386.rpm: size pgp md5 OK
+----
+
+[[Multiple-Builds]]
+==== Multiple Builds
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: This is missing a verb. Maybe "Building multiple packages"?

File: source/advanced-topics.adoc [hunk 11]
```diff
@@ -4,6 +4,187 @@
...
+
+When building multiple packages, use the following syntax to avoid entering the PGP passphrase multiple times. For example when building the blather and bother packages, sign them by using the following:
```
Comment (pmkovar) [inline source/advanced-topics.adoc]: Use literal for all package names.

File: source/advanced-topics.adoc (hunk 4 above)
Comment (Daskyleia) [reply source/advanced-topics.adoc]: This might be a question for @xsuchy or others...

Comment (xsuchy) [reply source/advanced-topics.adoc]: It should be run by the user who is adding the signature. So normal user.

**Pull request merged**